src/cursor.rs

```diff
@@ -0,0 +1,146 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CustomCursor {
+    pub(crate) inner: Arc<PlatformCustomCursor>,
+}
+
+impl CustomCursor {
+    pub fn from_rgba(
+        rgba: Vec<u8>,
+        width: u32,
+        height: u32,
+        hotspot_x: u32,
+        hotspot_y: u32,
+    ) -> Result<Self, BadImage> {
+        Ok(Self {
+            inner: PlatformCustomCursor::from_rgba(rgba, width, height, hotspot_x, hotspot_y)?
+                .into(),
+        })
+    }
+}
+
+/// Platforms that don't support cursors will export this as `PlatformCustomCursor`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct NoCustomCursor;
+
+#[allow(dead_code)]
+impl NoCustomCursor {
+    pub fn from_rgba(
+        rgba: Vec<u8>,
+        width: u32,
+        height: u32,
```

Comment:
Do we really need a `width` and `height`? I'd just use `stride: u32`, since you can internally derive the rest from it.

Reply:
Not really required, but works as a sanity check.

Reply:
But also I don't really care about it so I can change it.

Reply:
`stride` and `len` of the supplied buffer will result in the exact same behavior but with one less argument, though, `Icon` uses the `height` and `width`, so we could keep them as is.

Reply:
I know, but I guess we are keeping it then.